### ISSUE TITLE
Feature request: Add config for typescript

### ISSUE THREAD

**Issue by iliubinskii** [mysticatea/eslint-plugin-es#73]
Hello,

Thank you for the plugin.

I am using typescript with target=ES2017 option.

If I have:
```ts
const y = { ...x };
const re = /(?<!a)b/u;
```
It is transpiled:
```
const y = Object.assign({}, x);
const re = /(?<!a)b/u;
```

So, with typescript I can use spread operator, but not regexp lookback.
Your restrict-to-es2017 does not catch this.

My suggestion is to add configs for typescript that should disallow those features that are missing in respective ES version and not transpiled by typescript.

I think that typescript configs should include only regexp-related rules, but not sure.
